src/robot/model/visitor.py

```diff
@@ -159,10 +159,10 @@ def visit_test(self, test: TestCase):
         """
         if self.start_test(test) is not False:
             if test.has_setup:
-                test.setup.visit(self)
+                test.setup.visit(self)  # type: ignore
             test.body.visit(self)
             if test.has_teardown:
-                test.teardown.visit(self)
+                test.teardown.visit(self) # type: ignore
```

Comment:
I don't like `# type: ignore` comments in general, but in these cases they shouldn't even be needed.

Reply:
I will remove these. Unfortunately, most IDEs don't understand that there is a valid check beforehand.  has_teardown and has_setup will need to implement [typeguards](https://peps.python.org/pep-0647) but they are not available until the very latest versions of python

Reply:
`test.teardown` and others ought to have type hint `Keyword` so there shouldn't be errors calling `test.teardown.visit(self)`. At least earlier the type hint was `Keyword | None` which would have issues.

The `has_teardown` check is there to avoid accessing `test.teardown` property that creates the underlying `Keyword` object. It wouldn't be a problem otherwise, but creating lot of such objects unnecessarily consumes memory.